processor/coralogixprocessor/README.md

```diff
@@ -0,0 +1,70 @@
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+<!-- end autogenerated section -->
+
+## Description
+
+The Coralogix processor adds attributes to spans that enable features in Coralogix.
+
+## Features
+### DB Statement Blueprints
+This feature enables the processor to create blueprints from SQL queries, this means replacing any variables with `?`.
+The blueprint is also hashed to be able to be used with spanmetrics connector. 
+Long queries can be an issue when being stored in certain metric stores, 
+this way its possible to use the hash as the identifying dimension on the metric and be able the query certain blueprints.  
+
+The added attributes are `db.statement.blueprint` and `db.statement.blueprint.id`.
+* `db.statement.blueprint` contains the blueprinted version of the statement, we require them to be sent to coralogix to display your blueprinted statement
+* `db.statement.blueprint.id` contains a hash of the statement, this way we can add it as a dimension in the spanmetrics connector and use it to query your blueprints.
+* `sampling.priority` if enabled contains the value 100 for new blueprints, further explanation below.
+
+#### Sampling
+
+If sampling is enabled then it stores the found blueprints in an in-memory cache to be able to send only new blueprints that haven't been seen yet.
+This only adds an attribute to the span named `sampling.priority`, if the blueprint is new then the sampling priority will be `100`.
+
+Using this key it's possible to use either the Tail Sampler
+or the Probabilistic Sampler to only send new blueprints to coralogix.
+If sampling is not enabled it won't cache anything and the `sampling.priorty` attribute won't be added. 
+
+## Config
+* `db_statment_blueprints`
+  * `with_samping` (default: `false`): Enables cache to store seen blueprints and adds the attribute `sampling.priority` to spans with new blueprints.
+  * `cache_config`
+    * `max_cache_size_mib` (default: `1048576`) The size of the cache in mebibytes to store seen blueprints hashes.
+    * `max_cached_entries` (default: `15000000`) The number of seen blueprints the cache will store.
+
+### Basic Setup
+```yaml
+processors:
+  coralogix:
+    # Enables blueprints feature
+    db_statement_blueprints:
+      # Enables sampling, the cache will be with default values
+      with_sampling: true
+```
+
+### With Cache Config
+```yaml
+processors:
+  coralogix:
+    # Enables blueprints feature
+    db_statement_blueprints:
+      # Enables sampling, the cache will be with default values
+      with_sampling: true
+      # Custom cache configuration
+      cache_config:
+        # Maximum number of mebibytes stored in the cache
+        max_cache_size_mib: 1048576 #1GB
+        # Maximum number of hashes (IDs) stored in cache
+        max_cached_entries: 10000000 # 1 million
```

Comment:
```suggestion
processors:
  coralogix:
    db_statement_blueprints:
      sampling: 
        enabled: true
        max_cache_size_mib: 1048576 #1GB
        max_cached_entries: 10000000 # 1 million
```
Do you think having the cache options as a subsection under `sampling` would make the functionality dependency more clear?

The main concern I'd have with my suggestion is if the cache ever needs to be used for something else. I'll defer to you to know future plans here.